include/oneapi/dpl/experimental/kt/esimd_radix_sort_utils.h

```diff
@@ -489,34 +489,86 @@ VectorStore(uint32_t offset,
     return VectorStore<T, VSize, LANES>({offset, LaneStride*sizeof(T)}, data, mask);
 }
 
-template <typename T, int N>
-inline std::enable_if_t< (N*sizeof(T)/sizeof(uint32_t)<=64), __ESIMD_NS::simd<T, N> > BlockLoad(uint32_t slm_offset)
+template <int NElts>
+constexpr int
+lsc_op_block_size_rounding()
+{
+    static_assert(NElts >= 1, "");
+
+    if (NElts < 2)
+        return 1;
+
+    if (NElts < 3)
+        return 2;
+
+    if (NElts < 4)
+        return 3;
+
+    if (NElts < 8)
+        return 4;
+
+    if (NElts < 16)
+        return 8;
+
+    if (NElts < 32)
+        return 16;
+
+    if (NElts < 64)
+        return 32;
+
+    return 64;
+}
+
+template <typename SrcType, int N, typename DestType>
+constexpr int
+lsc_op_block_size
+{
+    return N * sizeof(SrcType) / sizeof(DestType);
+}
+
+template <typename T>
+using lsc_op_aligned_t = sizeof(T) <= sizeof(::std::uint32_t) ? ::std::uint32_t : ::std::uint64_t;
+
+template <typename T, int N, typename OpAlignedT = lsc_op_aligned_t<T>, int NElts = lsc_op_block_size<T, N, OpAlignedT>(),
+          ::std::enable_if_t<NElts == lsc_op_block_size_rounding<NElts>(), int> = 0>
+inline __ESIMD_NS::simd<T, N>
+BlockLoad(uint32_t slm_offset)
 {
     __ESIMD_NS::simd<T, N> result;
-    result.template bit_cast_view<uint32_t>() = __ESIMD_ENS::lsc_slm_block_load<uint32_t, N*sizeof(T)/sizeof(uint32_t)>(slm_offset);
+    result.template bit_cast_view<OpAlignedT>() = __ESIMD_ENS::lsc_slm_block_load<OpAlignedT, NElts>(slm_offset);
     return result;
 }
 
-template <typename T, int N>
-inline std::enable_if_t< (N*sizeof(T)/sizeof(uint32_t)>64), __ESIMD_NS::simd<T, N> > BlockLoad(uint32_t slm_offset)
+template <typename T, int N, typename OpAlignedT = lsc_op_aligned_t<T>, int NElts = lsc_op_block_size<T, N, OpAlignedT>(),
+          ::std::enable_if_t<NElts != lsc_op_block_size_rounding<NElts>(), int> = 0>
+inline __ESIMD_NS::simd<T, N>
+BlockLoad(uint32_t slm_offset)
 {
+    constexpr int ItemsCountsOfType_OpAlignedT = lsc_op_block_size_rounding<NElts>();
```

Comment:
`ItemsCountsOfType_OpAlignedT` is way too complicated... Could you use `BLOCK_SIZE_ROUNDED`?

Reply:
done.